### PR TITLE
Add of a new key/value pair in the json dictionnary. The key is messa…

### LINKED ISSUE #1512: JSON output format does not include "msg_id" field
### Steps to reproduce
1.   Run PyLint with output-format = json on any file that generates error or warning messages

### Current behavior
The returned results have no "msg_id" field, so the caller cannot process the results based on the message type which is unique to the msg_id, as opposed to the actual message text.

The currently returned fields are:  message, line, module, path, symbol, type, column, obj

### Expected behavior
Include the msg_id field in the returned JSON dictionaries by default, just as the msg_id is returned by default in the other output-formats.    

The impact on existing user code should be minimal for those that are using normal JSON decoders as the additional field will be ignored as an unused field in the resultant dictionary.

Additional note:  It would be nice if the "msg-template" option could be used to control the fields being returned by the json output-format just like it does for the other output-formats, e.g.  msg-template = '{{ "id":{msg_id}, "location": {obj}, "line":{line}, "msg":{message}" }}'

### pylint --version output

No config file found, using default configuration
pylint 1.7.1,
astroid 1.5.2
Python 3.5.3 (v3.5.3:1880cb95a742, Jan 16 2017, 16:02:32) [MSC v.1900 64 bit (AMD64)]

### PR DESCRIPTION
**Pull request by hippo91**
…ge-id and the value is the message id. closes #1512

### Fixes / new features
-

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -24,6 +24,9 @@ What's New in Pylint 1.8?
     * Added a new Python 3 check for declaring a method ``next`` that would have
       been treated as an iterator in Python 2 but a normal function in Python 3.
 
+		* Added a new key-value pair in json output. The key is ``message-id`` and
```
Comment (rogalski) [inline ChangeLog]: it seems that indentation is off. Please use spaces instead of tabs.

File: pylint/test/unittest_reporters_json.py [hunk 2]
```diff
@@ -34,6 +34,7 @@ def test_simple_json_output():
         ("column", 0),
         ("line", 1),
         ("message", "Line too long (1/2)"),
+        ("message-id", "C0301"),
```
Comment (rogalski) [inline pylint/test/unittest_reporters_json.py]: IIRC issue mentioned reporting _symbolic message_ - so in this case it would be `line-too-long`.

Comment (hippo91) [inline pylint/test/unittest_reporters_json.py]: And what about adding the message-id (C0301 for example) and the message-symbol (line-too-long)?

Comment (hippo91) [inline pylint/test/unittest_reporters_json.py]: In fact, the symbol is already packed in the dictionary. The key is ``symbol``.

File: doc/whatsnew/1.8.rst [hunk 3]
```diff
@@ -84,3 +84,6 @@ Summary -- Release highlights
         def foo(self):
             """int: My Numpy and Google docstring style description."""
             return 10
+
+* In case of ``--output-format=json``, the dictionnary returned holds a new key-value pair.
```
Comment (AWhetter) [inline doc/whatsnew/1.8.rst]: There's a typo here. "dictionnary" should be "dictionary".

**Comment by PCManticore** [comment]
Makes sense, thanks!

**Pull request merged**